src/Pos/Block/Worker.hs

```diff
@@ -155,30 +154,42 @@ onNewSlotWhenLeader slotId pSk sendActions = do
     logInfoS $
         sformat ("Waiting for "%shown%" before creating block") timeToWait
     delay timeToWait
-    let onNewSlotWhenLeaderDo = do
-            logInfoS "It's time to create a block for current slot"
-            let whenCreated createdBlk = do
-                    logInfoS $
-                        sformat ("Created a new block:\n" %build) createdBlk
-                    jlLog $ jlCreatedBlock (Right createdBlk)
-                    verifyCreatedBlock createdBlk
-                    void $ fork $ announceBlock sendActions $ createdBlk ^. gbHeader
-            let whenNotCreated = logWarningS . (mappend "I couldn't create a new block: ")
-            createdBlock <- createMainBlock slotId pSk
-            either whenNotCreated whenCreated createdBlock
-            logInfoS "onNewSlotWhenLeader: done"
     logWarningSWaitLinear 8 "onNewSlotWhenLeader" onNewSlotWhenLeaderDo
+  where
+    onNewSlotWhenLeaderDo = do
+        logInfoS "It's time to create a block for current slot"
+        createBlock verifyAndFallbackIfNeeded
+        logInfoS "onNewSlotWhenLeader: done"
+
+    createBlock onSuccess = do
+        let whenCreated createdBlk = do
+                logInfoS $
+                    sformat ("Created a new block:\n" %build) createdBlk
+                jlLog $ jlCreatedBlock (Right createdBlk)
+                onSuccess createdBlk
+        let whenNotCreated = logWarningS . (mappend "I couldn't create a new block: ")
+        createdBlock <- createMainBlock slotId pSk
+        either whenNotCreated whenCreated createdBlock
+
+    verifyAndFallbackIfNeeded createdBlk = case verifyCreatedBlock createdBlk of
+        VerFailure errors -> do
+            logWarning $ sformat ("New block failed some checks: "%listJson) errors
```

Comment:
I used `logError`, but thought that `logError` means that error is fatal and application must be aborted.